src/raftstore/store/store.rs

```diff
@@ -3136,6 +3141,49 @@ impl<T: Transport, C: PdClient> Store<T, C> {
             error!("{} register cleanup import sst tick err: {:?}", self.tag, e);
         }
     }
+
+    /// Find the first region `r` that satisfies `r.end_key` > `from_key` and the peer on this TiKV
+    /// satisfies `filter(peer)` returns true.
+    fn seek_region(
+        &self,
+        from_key: &[u8],
+        filter: SeekRegionFilter,
+        mut limit: u32,
+        callback: SeekRegionCallback,
+    ) {
+        // `limit` should be at least 1
+        if limit == 0 {
+            limit = 1;
+        }
+
+        let from_key = data_end_key(from_key);
```

Comment:
When from_key is empty, it will should return the first region.

Reply:
It finds region accroding to it's end_key. The minimal end_key is `"\x00"`, and the max end_key was represented as empty.
Although, the `data_end_key` function adds `'z'` prefix to a key, or returns `'z' + 1` if the given key is empty.

Reply:
I believe it will be much more elegant if we can search region by start_key...

Reply:
You don't get my point. Seek in RocksDB means find the smallest key equal to or larger than the given key. Here seek region should return the first region whose range cover or larger than the given key. Don't mix the way we encoding key with this concept.

Reply:
Here you should use `let from_key = data_key(from_key)`.